src/openforms/variables/validators.py

```diff
@@ -123,6 +123,51 @@ def __call__(self, value: Mapping[str, str] | JSONValue) -> None:
             raise ValidationError(errors)
 
 
+@deconstructible
+class QueryParameterValidator:
+    """Validates if the value is a mapping of strings to lists of strings"""
+
+    def __call__(self, value: Mapping[str, list[str]] | None) -> None:
+        if value is None:
+            return
+        if not isinstance(value, Mapping):
+            raise ValidationError(
+                _(
+                    'Query parameters should have the form {"parameter": ["my", "parameter", "values"]}'
+                )
+            )
+
+        errors = []
+        for field_name, field_value in value.items():
+            # value checks
+            if not isinstance(field_value, list):
+                errors.append(
+                    _(
+                        "{parameter!s}: value '{value!s}' should be a list, but isn't."
+                    ).format(
+                        parameter=field_name,
+                        value=field_value,
+                    )
+                )
+            elif len(field_value):
+                if not all(isinstance(sub_value, str) for sub_value in field_value):
+                    errors.append(
+                        _(
+                            "{parameter!s}: value '{value!s}' should be a list of strings, but isn't."
```

Comment:
the exact messages here defined are not the same as the ones used in tests, so after entering translations the tests will break. Please use the exact same messages.